saleor/graphql/product/bulk_mutations/products.py

```diff
@@ -693,34 +693,71 @@ class Meta:
             info, data["variant_id"], only_type=ProductVariant
         )
-        if stocks:
-            warehouse_ids = [stock["warehouse"] for stock in stocks]
+        if stocks_input:
+            warehouse_ids = [stock["warehouse"] for stock in stocks_input]
             cls.check_for_duplicates(warehouse_ids, errors)
             if errors:
                 raise ValidationError(errors)
             warehouses = cls.get_nodes_or_error(
                 warehouse_ids, "warehouse", only_type=Warehouse
             )
-            cls.update_or_create_variant_stocks(variant, stocks, warehouses)
+            cls.update_or_create_variant_stocks(variant, stocks_input, warehouses, info)
 
         variant = ChannelContext(node=variant, channel_slug=None)
         return cls(product_variant=variant)
 
     @classmethod
     @traced_atomic_transaction()
-    def update_or_create_variant_stocks(cls, variant, stocks_data, warehouses):
-        stocks = []
-        for stock_data, warehouse in zip(stocks_data, warehouses):
-            stock, _ = warehouse_models.Stock.objects.get_or_create(
+    def update_or_create_variant_stocks(cls, variant, stocks_input, warehouses, info):
+        stocks_data = []
+        for stock_data, warehouse in zip(stocks_input, warehouses):
+            stock, created = warehouse_models.Stock.objects.get_or_create(
                 product_variant=variant, warehouse=warehouse
             )
+            stocks_data.append(
+                {
+                    "stock": stock,
+                    "created": created,
+                    "prev_quantity": stock.quantity,
+                    "current_quantity": stock_data["quantity"],
+                }
+            )
             stock.quantity = stock_data["quantity"]
-            stocks.append(stock)
-        warehouse_models.Stock.objects.bulk_update(stocks, ["quantity"])
+        stocks_to_update = [data["stock"] for data in stocks_data]
+        warehouse_models.Stock.objects.bulk_update(stocks_to_update, ["quantity"])
+        plugins_manager = info.context.plugins
+        cls._run_product_variant_back_in_stock_webhook(plugins_manager, stocks_data)
+        cls._run_product_variant_stock_changed_webhook(
+            plugins_manager, stocks_to_update
+        )
+
+    @classmethod
+    def _run_product_variant_back_in_stock_webhook(cls, plugins_manager, stocks_data):
+        for stock in (
+            data["stock"]
+            for data in stocks_data
+            if cls._product_variant_webhook_conditions(data)
+        ):
+            plugins_manager.product_variant_back_in_stock(stock.product_variant)
+
+    @classmethod
+    def _run_product_variant_stock_changed_webhook(
+        cls, plugins_manager, updated_stocks
+    ):
+        for stock in updated_stocks:
+            plugins_manager.product_variant_stock_changed(stock.product_variant)
+
+    @classmethod
+    def _product_variant_webhook_conditions(cls, updated_stock):
```

Comment:
I would rename this function so that the name suggests what it checks, e.g. `_is_variant_back_in_stock`